### ISSUE TITLE
Non deterministically behaviour of date and number recognition

### ISSUE THREAD

**Issue by DankessS** [vic64/nlpcraft#57]
I am using NLPCraft 0.5.1. I ask query:
`Twitter for \"Qatar\" since 01.01.2018 to 01.01.2019`

NLPCraft recognizes it as:

```
tokens = {Wrappers$SeqWrapper@10948}  size = 4
 0 = {NCTokenImpl@10974} "Token [id=x:twitterFor, text=twitter, group=x:twitterFor, value=null]"
 1 = {NCTokenImpl@10975} "Token [id=nlp:nlp, text=for, group=nlp:nlp, value=null]"
 2 = {NCTokenImpl@10976} "Token [id=nlp:geo, text=qatar, group=nlp:geo, value=null]"
 3 = {NCTokenImpl@10977} "Token [id=nlp:num, text=since 01.01.2018 to 01.01.2019, group=nlp:num, value=null]"
```

but when I ask:
`Twitter for \"Qatar\" since 01.01.2018 to 01.01.2019 limit 100` 
or
`Twitter for \"Qatar\" limit 100 since 01.01.2018 to 01.01.2019`
then tokens are:
```
tokens = {Wrappers$SeqWrapper@11009}  size = 6
 0 = {NCTokenImpl@11011} "Token [id=x:twitterFor, text=twitter, group=x:twitterFor, value=null]"
 1 = {NCTokenImpl@11012} "Token [id=nlp:nlp, text=for, group=nlp:nlp, value=null]"
 2 = {NCTokenImpl@11013} "Token [id=nlp:geo, text=qatar, group=nlp:geo, value=null]"
 3 = {NCTokenImpl@11014} "Token [id=nlp:date, text=since 01.01.2018 to 01.01.2019, group=nlp:date, value=null]"
 4 = {NCTokenImpl@11015} "Token [id=nlp:nlp, text=limit, group=nlp:nlp, value=null]"
 5 = {NCTokenImpl@11016} "Token [id=nlp:num, text=100, group=nlp:num, value=null]"
```

I don't understand why sentence `since 01.01.2018 to 01.01.2019` in first example is recognized as `nlp:num` and in second as `nlp:date`. Is it connected with date format? When I use MM-DD-YYYY instead of MM.DD.YYYY everythink works fine. But it is a little confusing while using DD.MM.YYYY format. Please check it.

**Comment by skhdl** [comment]
Fix for review ready GH-57
1) Strings like "01.01.2018" aren't parsed as numbers now (note that like "01.011.2018" parsed as numbers)
2) Note that string like "01.2018" parsed as a date and a number. So intent should expect one of these types (variants list will contain both of them).

**Comment by skhdl** [comment]
@DankessS 
Could you tell us which intents do you use? 
The current version can incorrectly detect some dates as numerics, but it should add only some redundant variant, but it shouldn't influence on correct intent with expected dates.

**Comment by DankessS** [comment]
@skhdl  Thanks. I forgot to add that I'm not using intent with expected dates or numbers. I rely on nlpcraft recognition and when dates appear in intent solver then I take them into accout, otherwise not.

**Comment by skhdl** [comment]
@DankessS 
We'll add our fix into next release (strings like `01.01.2018` will not be parsed as numbers)
But please take into account that better to expect some optional dates in your intent because strings like '10.10' will be recognized as dates and as numbers (2 variants for such cases) and if you expect for example some optional date in your intent - variants with numbers will not bother you.
So, you can solve your problem even before our next release fix.

**Comment by aradzinski** [comment]
@DankessS if would be great if you can share your intents with us so we can quickly help. If necessary, you can send it to support@nlpcraft.org to keep it confidential.

**Comment by aradzinski** [comment]
Fix merged to 'release' branch. Keep it open for now...

**Comment by DankessS** [comment]
@aradzinski @skhdl  I made some workaround in my code, so I handled this problem properly. I can calmly wait for next nlpcraft release, I'm in no hurry now. Thanks for advices.

**Comment by DankessS** [comment]
@aradzinski @skhdl I don't know if this was fixed already, but I get wrong recognition in NLPCraft 0.5.1 when asking question:
`... since 08/25/2000`
token "08/25/2000" is recognized as nlp:nlp, the same situation is for all dates between year 2010, for example 08/25/2005, 08/25/2009, but when type 08/25/2010, 08/25/2011 and so on, then token is recognized as nlp:date.  I would like to notice that I have set MDY date format in application.conf and I expect date in intent solver in this way 
`new TERM("id == nlp:date", 0, 1)`
so it is optional, but when typed in query I want it to be recognized as date.
Please check this case. If you need more info, then inform me, I will deliver it.
When do you plan to release NLPCraft 0.6.0 ?

**Comment by aradzinski** [comment]
@DankessS the fix is part of upcoming 0.6.0 release. Should be in the next couple of days.

**Comment by DankessS** [comment]
Ok, so I am creating new issue #77
